b1ddi/util/util.go

```diff
@@ -0,0 +1,61 @@
+package util
+
+import (
+	"errors"
+	"fmt"
+	"runtime/debug"
+	"strconv"
+)
+
+var (
+	ParseError = "unable to parse key '%s': %v"
+	goSDKPath  = "github.com/infobloxopen/b1ddi-go-client"
+)
+
+func ToInt(d map[string]interface{}, key string) (int, bool, error) {
```

Comment:
Why do you need bool (ok) value, if you have an error, which you can always check for `nil` value? 

Please simplify this
```suggestion
func ToInt(d map[string]interface{}, key string) (int, error) {
```

Reply:
I need the boolean value because the key could be optional.
Not all the keys in the DNS Record option and rData are required. Consider this scenario: if let's say the key doesn't exist, the error would be nil and w/o this boolean I would have a way to tell if the update was required or not in the calling function.